### PR TITLE
[generator] Store which XM profile we're targeting and use it to determine if we have a reference to System.Drawing or not.

### PR DESCRIPTION
**Pull request by rolfbjarne**

### CODE REVIEW

**Comment by rolfbjarne** [comment]
@chamons Can you have a look at this and see if it looks good to you?

File: src/btouch.cs [hunk 1]
```diff
@@ -224,8 +236,8 @@ static int Main2 (string [] args)
 					linkwith.Add (id);
 				}
 			},
-			{ "unified-full-profile", "Launches compiler pointing to XM Full Profile", l => { /* no-op*/ }, true },
```
Comment (chamons) [inline src/btouch.cs]: Love the idea, but I chose a different name here:

https://github.com/xamarin/xamarin-macios/blob/master/tools/mmp/driver.cs#L362

I kinda like yours better. I can PR after this goes in swapping the name unless you want to add it here.

**Comment by chamons** [comment]
👍

File: src/btouch.cs (hunk 1 above)
Comment (rolfbjarne) [inline src/btouch.cs]: I think I have a better idea, we can extend our use of TargetFramework to use a TargetFrameworkMoniker: https://blogs.msdn.microsoft.com/visualstudio/2010/05/19/visual-studio-managed-multi-targeting-part-1-concepts-target-framework-moniker-target-framework/

So we'd end up with:

    --target-framework=Xamarin.Mac,Version=v4.5,Profile=Full
    --target-framework=Xamarin.Mac,Version=v4.5,Profile=System
    --target-framework=Xamarin.Mac,Version=v2.0,Profile=Mobile

as the possible arguments.

Comment (rolfbjarne) [inline src/btouch.cs]: OK, I'll change my PR.

**Comment by rolfbjarne** [comment]
This is continued in PR #1339.

**Pull request closed without merging**